finality-aleph/src/party/mod.rs

```diff
@@ -435,6 +449,32 @@ pub(crate) fn create_aleph_config(
     consensus_config
 }
 
+/// Removes the backup directory for a session.
+///
+/// `backup_path` is the path to the backup directory (i.e. the argument to `--backup-saving-path`).
+/// If it is `None`, nothing is done.
+///
+/// Any filesystem errors are logged and dropped.
+///
+/// This should be done after the end of the session.
+fn remove_session_backup(backup_path: Option<PathBuf>, session_id: u32) {
+    let path = match backup_path {
+        Some(path) => path.join(session_id.to_string()),
+        None => return,
+    };
+    match fs::remove_dir_all(&path) {
+        Ok(()) => {
+            debug!(target: "aleph-party", "Removed backup for session {}", session_id);
+        }
+        Err(error) => {
+            let dir_exists = || !matches!(fs::metadata(&path), Err(error) if error.kind() == io::ErrorKind::NotFound);
+            if error.kind() != io::ErrorKind::NotFound || dir_exists() {
```

Comment:
This logic seems unnecessarily convoluted. Could we just check whether the directory still exists, and log the error if it does? This is the exact worrying situation anyway.

To be precise I owuld just put the body of `dir_exists` in this `if`.

Reply:
Hmm, maybe just printing warning with error message? I don't think this distinction gives us anything more?

Reply:
Oh, no, that's not what I had in mind. The thing we care about here is whether the dir still exists after the removal attempt, so we should only log any problems when it does.

The current version has the problem that it `warn`s when the directory wasn't there, which afaik can very well be the case during normal operation.

Reply:
Oh, so try to remove, ignore any errors, then call `metadata()`, and make sure it's `NotFound` (not `Ok` and not another error)? [Like this](https://github.com/Cardinal-Cryptography/aleph-node/commit/07e45e1)?

Reply:
I would keep the original error for logging in case the dir is still there, but yeah. Also, maybe only check this if the removal returned an error, the case where it didn't and the dir exists seems overly pessimistic even for me. :D

Reply:
I'm pretty sure this removal also happens e.g. after sessions in which the node is not an authority, in which case it did not create these dirs. :/

Reply:
There also might be a problem if someone restarts the node after member fails early, backup gets deleted and everything restarts with empty backup. Maybe backup should be removed on start of the next session ://

Reply:
I like the last idea, but then it definitely shouldn't error on `NotFound`.

Reply:
Ok, I will move backup removing at start of session and remove warning on `NotFound`.